drivers/vulkan/rendering_device_driver_vulkan.cpp

```diff
@@ -2713,6 +2713,18 @@ RDD::PipelineID RenderingDeviceDriverVulkan::render_pipeline_create(
 	rasterization_state_create_info.depthBiasSlopeFactor = p_rasterization_state.depth_bias_slope_factor;
 	rasterization_state_create_info.lineWidth = p_rasterization_state.line_width;
 
+	// Hardware bresenham
+	VkPipelineRasterizationLineStateCreateInfoEXT rasterization_line_state_create_info_ext;
+	rasterization_line_state_create_info_ext.sType = VK_STRUCTURE_TYPE_PIPELINE_RASTERIZATION_LINE_STATE_CREATE_INFO_EXT;
+	rasterization_line_state_create_info_ext.pNext = nullptr;
+	// this line is ridiculous because the extension mimics the limitations of the original OpenGL line rasterization implementation
+	rasterization_line_state_create_info_ext.lineRasterizationMode = ((Math::is_zero_approx(p_rasterization_state.line_width) || p_rasterization_state.line_width == 1.0f) && !(p_dynamic_state.has_flag(DYNAMIC_STATE_LINE_WIDTH) || p_multisample_state.enable_alpha_to_coverage || p_multisample_state.enable_alpha_to_one || p_multisample_state.enable_sample_shading) ? VK_LINE_RASTERIZATION_MODE_BRESENHAM_EXT : VK_LINE_RASTERIZATION_MODE_DEFAULT_EXT);
```

Comment:
```suggestion
	rasterization_line_state_create_info_ext.lineRasterizationMode = ((Math::is_zero_approx(p_rasterization_state.line_width) || p_rasterization_state.line_width == 1.0f) && !p_dynamic_state.has_flag(DYNAMIC_STATE_LINE_WIDTH) && !p_multisample_state.enable_alpha_to_coverage && !p_multisample_state.enable_alpha_to_one && !p_multisample_state.enable_sample_shading) ? VK_LINE_RASTERIZATION_MODE_BRESENHAM_EXT : VK_LINE_RASTERIZATION_MODE_DEFAULT_EXT;
```
Cleaner and more readable IMO